File: src/class/Money.js
import { messages } from "../utils/messages.js";

export class Money {
  balance = 0;

  constructor() {}

  validateAmount(amount) {
    if (isNaN(amount) || amount <= 0 || amount % 100 !== 0)
      throw new Error(messages.bad_amount_value);
  }

  validateAmountToWithdrawal(amount) {
    this.validateAmount(amount);
    if (amount > this.balance) throw new Error(messages.insufficient_balance);
  }

  getBalance = () => `${this.balance} Ar`;

  deposit(amount) {
    this.validateAmount(+amount);
    this.balance += +amount;
  }

  withdrawal(amount) {
    this.validateAmountToWithdrawal(+amount);
    this.balance -= +amount;
  }
}

File: src/class/money.dart
import '../utils/messages.dart';

class Money {
  int balance = 0;

  void validateAmount(int amount) {
    if (amount.isNaN || amount <= 0 || amount % 100 != 0) {
      throw Exception(messages['bad_amount_value']);
    }
  }

  void validateAmountToWithdrawal(int amount) {
    validateAmount(amount);
    if (amount > balance) {
      throw Exception(messages['insufficient_balance']);
    }
  }

  String getBalance() => '$balance Ar';

  void deposit(int amount) {
    validateAmount(amount);
    balance += amount;
  }

  void withdrawal(int amount) {
    validateAmountToWithdrawal(amount);
    balance -= amount;
  }
}
